src/devices_models/devices/thermal_generation.jl

```diff
@@ -301,6 +349,7 @@ function add_constraints!(
         names,
         time_steps,
         meta = "off",
+        sparse = true,
```

Comment:
Why is this a sparse container?

Reply:
that constraint is not created for the last time step, so the container had a undef. I should have changed the axes to be `time steps[1:end-1]` then we can avoid the usage of spare. sorry my bad.